File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.java
class Solution {
    public int maxDistance(int[] colors) {
        int answer=0;
        for(int left=0;left<colors.length;left++){
            for(int right=left+1;right<colors.length;right++){
                if(colors[right]!=colors[left]){
                    answer=Math.max(answer,right-left);
                }
            }
        }

        return answer;
    }
}

File: 2078-two-furthest-houses-with-different-colors/2078-two-furthest-houses-with-different-colors.py
class Solution:
    def maxDistance(self, colors: List[int]) -> int:
        answer = 0
        n = len(colors)
        for i in range(n):
            for j in range(n):
                if colors[i] != colors[j]:
                    answer = max(answer, j - i)

        return answer

        